src/lib.rs

```diff
@@ -4221,6 +4263,161 @@ fn map_darwin_target_from_rust_to_compiler_architecture(target: &Target) -> &str
     }
 }
 
+// Rust and clang/cc don't agree on what equivalent flags should look like either.
+fn rustc_to_cc_flag(flag: &str, family: ToolFamily) -> Option<OsString> {
+    match family {
+        ToolFamily::Clang { .. } | ToolFamily::Gnu => match flag {
+            _ if flag.starts_with("branch-protection=") => {
+                Some(format!("-m{}", flag.replace(",", "+")).into())
+            }
+            _ if flag.starts_with("code-model=") => {
+                Some(flag.replace("code-model", "-mcmodel").into())
+            }
+            _ if flag.starts_with("no-vectorize-loops") => Some("-fno-vectorize".into()),
+            _ if flag.starts_with("no-vectorize-slp") => Some("-fno-slp-vectorize".into()),
+            _ if flag.starts_with("profile-generate") => Some(format!("-f{}", flag).into()),
```

Comment:
I'd prefer inline arg if our msrv supports it:

```suggestion
            _ if flag.starts_with("profile-generate") => Some(format!("-f{flag}").into()),
```